app/controllers/SeriesDetailsController.scala

```diff
@@ -1,33 +1,46 @@
 package controllers
 
+import auth.TokenSecurity
+import configuration.{GraphQLConfiguration, KeycloakConfiguration}
+import graphql.codegen.GetSeries.getSeries._
 import javax.inject.{Inject, Singleton}
-import org.pac4j.core.profile.CommonProfile
-import org.pac4j.play.scala.{Security, SecurityComponents}
+import org.pac4j.play.scala.SecurityComponents
 import play.api.data.Form
 import play.api.data.Forms._
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, Request}
 
+import scala.concurrent.ExecutionContext
+
 @Singleton
-class SeriesDetailsController @Inject()(val controllerComponents: SecurityComponents)
-                                         extends Security[CommonProfile] with I18nSupport  {
+class SeriesDetailsController @Inject()(val controllerComponents: SecurityComponents,
+                                        val graphqlConfiguration: GraphQLConfiguration,
+                                        val keycloakConfiguration: KeycloakConfiguration
+                                        )(implicit val ec: ExecutionContext) extends TokenSecurity with I18nSupport {
 
   private val secureAction = Secure("OidcClient")
+  private val client = graphqlConfiguration.getClient[Data, Variables]()
 
   val selectedSeriesForm = Form(
     mapping(
       "seriesId" -> nonEmptyText
     )(SelectedSeriesData.apply)(SelectedSeriesData.unapply)
   )
 
-  // Mocked data until API implemented to show dropdown on page working
-  val mockedAllSeriesData: Seq[(String, String)] = Seq(
-    ("mockedSeries1", "Mocked Series 1"),
-    ("mockedSeries2", "Mocked Series 2"),
-    ("mockedSeries3", "Mocked Series 3"))
+  def seriesDetails(): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
+    val body = keycloakConfiguration.verifyToken(request.token.getValue)
```

Comment:
What do you think about naming this variable something like `userParentBody` or `userTransferringBody`? It took me a minute to realise that `body` was a department rather than the HTTP request body.

Reply:
`userTransferringBody` sounds right for most use cases. I've changed it to that here.